the-collab-lab/tcl-22-smart-shopping-list
# Fix Navigation Errors when Token is Not Set

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



Currently, the app allows the user to navigate to /list and /additems even if they don't have a token yet. This PR fixes it in two ways:
* it removes the `<nav>` from the welcome screen, so the user doesn't see any links until a token is created
* it uses router to redirect to the welcome screen if the user tries to manually navigate to /list and /additems without a token.

## Related Issue


Closes #23 

## Acceptance Criteria



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓  | :bug: Bug fix              |

## Updates

### Before


https://www.loom.com/share/fc4997267b564ca0b5627c7d78fc9f5f

### After


https://www.loom.com/share/1ec10b9ff2c14f8a9016d07013061ac8


## Testing Steps / QA Criteria


To test the feature, use one of the following ways:

1. Follow the [preview link](https://tcl-22-shopping-list--pr54-dt-no-token-no-nav-rx3dcp1x.web.app)
2. Launch the app locally from your terminal:
- Pull down this branch with `git pull origin dt-no-token-no-nav` and check that branch out with `git checkout origin dt-no-token-no-nav`
- Run npm start to launch the app in your browser at http://localhost:3000

In the app:

1. Clear your localStorage.
2. Navigate to the main page.
3. You should see the welcome screen and no navigation.
4. If you try to manually open /list or /additems, you will be redirected to the welcome screen.
5. After you create a new list or join an existing list, the navigation should work normally; the main page will redirect you to /list, as before.
